pandas/tests/indexing/test_loc.py

```diff
@@ -2085,6 +2088,60 @@ def test_loc_getitem_slice_columns_mixed_dtype(self):
         tm.assert_frame_equal(df.loc[:, 1:], expected)
 
 
+class TestLocBooleanLabelsAndSlices(Base):
+    @pytest.mark.parametrize(
+        "index",
+        [
+            RangeIndex(4),
+            Int64Index(range(4)),
+            UInt64Index(range(4)),
+            pd.Float64Index(range(4)),
+            CategoricalIndex(range(4)),
+            date_range(0, periods=4, freq="ns"),
+            timedelta_range(0, periods=4, freq="ns"),
+            pd.interval_range(0, periods=4),
+            Index([0, 1, 2, 3], dtype=object),
+            MultiIndex.from_product([[0, 1], [0, 1]]),
+            pd.period_range("2018Q1", freq="Q", periods=4),
+        ],
+    )
+    def test_loc_bool_incompatible_index_raises(self, index, frame_or_series):
+        # GH20432
+        message = "Boolean label can not be used without a boolean index"
+        obj = frame_or_series(range(4), index=index)
+        with pytest.raises(TypeError, match=message):
+            obj.loc[True]
+
+    @pytest.mark.parametrize("index", [Index([True, False], dtype="boolean")])
+    def test_loc_bool_should_not_raise(self, index, frame_or_series):
+        obj = frame_or_series(range(2), index=index)
+        obj.loc[True]
+
+    @pytest.mark.parametrize(
+        "index",
+        [
+            RangeIndex(4),
+            Int64Index(range(4)),
+            UInt64Index(range(4)),
+            pd.Float64Index(range(4)),
+            CategoricalIndex(range(4)),
+            date_range(0, periods=4, freq="ns"),
+            timedelta_range(0, periods=4, freq="ns"),
+            pd.interval_range(0, periods=4),
+            Index([0, 1, 2, 3], dtype=object),
+            Index([True, True, False, False], dtype=object),
+            MultiIndex.from_product([[0, 1], [0, 1]]),
+            pd.period_range("2018Q1", freq="Q", periods=4),
+        ],
```

Comment:
We probably have a fixture for the different index types?

Reply:
yep fixture is 'index'!

Reply:
Solved!